src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,104 @@
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DBALException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use ProxyManager\Proxy\LazyLoadingInterface;
+use Symfony\Bridge\Doctrine\Event\ForceKernelRebootEvent;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\VarExporter\LazyObjectInterface;
+
+/**
+ * Based on https://github.com/Baldinof/roadrunner-bundle/blob/3.x/src/Integration/Doctrine/DoctrineORMMiddleware.php.
+ */
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    public function __construct(private ManagerRegistry $managerRegistry, private ContainerInterface $container, private EventDispatcherInterface $eventDispatcher)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event)
+    {
+        $connectionServices = $this->managerRegistry->getConnectionNames();
+
+        foreach ($connectionServices as $connectionServiceName) {
+            if (!$this->container->initialized($connectionServiceName)) {
+                continue;
+            }
+
+            $connection = $this->container->get($connectionServiceName);
+
+            if (!$connection instanceof Connection) {
+                throw new \RuntimeException(sprintf('Is not an instance of "%s".', Connection::class));
+            }
+
+            if ($connection->isConnected() && false === $this->ping($connection)) {
+                $connection->close();
+            }
+
+            $managerNames = $this->managerRegistry->getManagerNames();
+
+            foreach ($managerNames as $managerName) {
+                if (!$this->container->initialized($managerName)) {
+                    continue;
+                }
+
+                $manager = $this->container->get($managerName);
+
+                if (!$manager instanceof EntityManagerInterface) {
+                    throw new \RuntimeException(sprintf('Is not an instance of "%s".', EntityManagerInterface::class));
+                }
+
+                if ($manager instanceof LazyLoadingInterface) {
+                    continue; // Doctrine bundle will handle manager reset on next request
+                }
+
+                if ($manager instanceof LazyObjectInterface) {
+                    continue; // Doctrine bundle will handle manager reset on next request
+                }
+
+                if (!$manager->isOpen()) {
+                    $this->eventDispatcher->dispatch(new ForceKernelRebootEvent(
+                        "entity manager '$managerName' is closed and the package `symfony/proxy-manager-bridge` is not installed so kernel reset will not re-open it"
```

Comment:
```suggestion
                        "Entity manager '$managerName' is closed and the package `symfony/proxy-manager-bridge` is not installed so kernel reset will not re-open it."
```

Maybe `sprintf` should be used in order to use double quotes around the manager name. This is what's used in other messages of the framework